src/AWS.Deploy.Orchestration/Utilities/Helpers.cs

```diff
@@ -0,0 +1,90 @@
+        /// <param name="frequency">Interval between the two executions of the task</param>
+        /// <param name="timeout">Interval for timeout, if timeout passes, methods throws <see cref="TimeoutException"/></param>
+        /// <exception cref="TimeoutException">Throws when timeout passes</exception>
+        public static async Task WaitUntil(Func<Task<bool>> predicate, TimeSpan frequency, TimeSpan timeout, CancellationToken cancellationToken = default)
+        {
+            var waitTask = Task.Run(async () =>
+            {
+                while (!cancellationToken.IsCancellationRequested && !await predicate())
+                {
+                    await Task.Delay(frequency);
+                }
+            });
+
+            if (waitTask != await Task.WhenAny(waitTask, Task.Delay(timeout)))
+            {
+                throw new TimeoutException();
+            }
+        }
+
+        /// <summary>
+        /// This method returns the deployment tool workspace directory to create the CDK app during deployment.
+        /// It will first look for AWS_DOTNET_DEPLOYTOOL_WORKSPACE environment variable set by the user. It will be used as the deploy tool workspace if it points to a valid directory whithout whitespace characters in its path.
+        /// If the environment variable is set, it will also create a temp directory inside the workspace and set process scoped TEMP and TMP environment variables that point to the temp directory.
+        /// This additional configuration is required due to a known issue in the CDK - https://github.com/aws/aws-cdk/issues/2532
+        /// If the override is not present, then it defaults to USERPROFILE/.aws-dotnet-deploy.
+        /// It will throw an exception if the USERPROFILE contains a whitespace character.
+        /// </summary>
+        public static string GetDeployToolWorkspaceDirectoryRoot(string userProfilePath, IDirectoryManager directoryManager, IEnvironmentVariableManager environmentVariableManager)
+        {
+            var overridenWorkspace = environmentVariableManager.GetEnvironmentVariable("AWS_DOTNET_DEPLOYTOOL_WORKSPACE", EnvironmentVariableTarget.User)
+               ?? environmentVariableManager.GetEnvironmentVariable("AWS_DOTNET_DEPLOYTOOL_WORKSPACE", EnvironmentVariableTarget.Machine);
+
+            var defaultWorkSpace = Path.Combine(userProfilePath, ".aws-dotnet-deploy");
+
+            if (overridenWorkspace == null)
+            {
+                return !defaultWorkSpace.Contains(" ")
+                    ? defaultWorkSpace
+                    : throw new InvalidDeployToolWorkspaceException(DeployToolErrorCode.InvalidDeployToolWorkspace, $"The USERPROFILE path ({userProfilePath}) contains a whitespace character and cannot be used as a workspace by the deployment tool. " +
+                    "Please refer to the troubleshooting guide for additional configuration required to handle this issue.");
```

Comment:
Can we be more prescriptive here and tell them refer to the troubleshooting guide for setting the `AWS_DOTNET_DEPLOYTOOL_WORKSPACE` environment variable to use an alternative workspace error.

Reply:
Done